### ISSUE TITLE
Should we strengthen the tests for the documented docker-compose.yml?

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#41395]
In #41394 we fixed the docs by adding an entry for `discovery.seed_hosts` to the recommended `docker-compose.yml` file in the reference docs. The issue would only appear when starting a cluster using `docker-compose` the _second_ time, because starting a cluster afresh is somewhat different from restarting it after a shutdown. This wasn't caught by our tests. Should we have tests to catch this kind of issue in future and, if so, how?

**Comment by not-napoleon** [comment]
We discussed this in fixit Thursday on 2019-04-25.  Our feeling is that there is value in having tests on the `docker-compose.yml` files, just to make sure they do what they're supposed to do in the happy path.  However, we felt that adding complex scenarios to catch errors we've already seen has rapidly diminishing returns. Something like a restart test here feels like an expensive regression test, which we didn't think added much value.  If there were a proposal for a test that would catch a class of errors, and not just one specific error we already encountered, that would be worth looking into, but tests that catch single, known issues doesn't provide enough utility for how fragile/expensive it would be.
